projects/graphql-client/src/mutation-tracker/mutation-tracker.service.ts

```diff
@@ -0,0 +1,75 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({ providedIn: 'root' })
+export class MutationTrackerService {
+  /**
+   * This tracker service assumes that a mutation affects two types of queries.
+   *  1. Query for mutated object by ID.
+   *  2. List query that contains the mutated object.
+   *
+   * - Correlation between the object/list is made via the MutationType construct.
```

Comment:
nit: naming - it's really just the object type, right? Wouldn't this just use the typename from gql? That would probably work in some cases - likely update (which usually takes in the same type as a query returns). Delete often just takes an ID though.